src/init.cpp

```diff
@@ -624,9 +624,22 @@ void ThreadStartWalletNotifier()
 
         if (walletBestBlockHash.has_value()) {
             int64_t slept{0};
+            bool isReindexing{true};
             auto timedOut = [&]() -> bool {
-                MilliSleep(50);
-                slept += 50;
+                MilliSleep(500);
+                {
+                    // re-check whether we're reindexing
+                    if (isReindexing) {
+                        LOCK(cs_main);
+                        pblocktree->ReadReindexing(isReindexing);
+                    }
+
+                    // once we're out of reindexing, we can start incrementing the slept counter
+                    if (!isReindexing) {
+                        slept += 500;
+                    }
+                }
```

Comment:
This could be unindented.

Reply:
Yeah, I was going to edit my earlier suggestion to note that, but you were too fast for me :-)